ravioli/ravioli.py

```diff
@@ -26,6 +26,10 @@ def report_all_functions(filename, args):
         # This is a directory. Run on all the files we can find.
         source_files = list(Path(filename).glob('**/*.c'))
 
+        if args.x:
+            for ext in args.x:
+                source_files += list(Path(filename).glob('**/*.' + str(ext)))
```

Comment:
It looks like this will run on all .c files _and_ whatever extensions you provide. I think the clearer behavior would be to default to C files, or only on any provided extensions with the `-x` option.